### DISCUSSION TITLE
graphql/codegen/docs/custom-codegen/plugin-structure

### DISCUSSION THREAD

**Discussion by giscus** [dotansimha/graphql-code-generator#9530] [Docs Discussions]
# graphql/codegen/docs/custom-codegen/plugin-structure

Learn how to write your own GraphQL Code Generator Plugin. You can use it to generate custom output, or to extend the existing plugins.

https://the-guild.dev/graphql/codegen/docs/custom-codegen/plugin-structure

**Comment by kartikjindal** [comment]
Can I get the output of the codegen script in my custom plugin?

**Reply by zecka** [reply]
@kartikjindal
If you mean the file content generated by codegen maybe you can use another approach than plugin.
You can use hooks on code

```ts
// codegen.ts
import { CodegenConfig } from '@graphql-codegen/cli';

export const config: CodegenConfig = {
    ...
    hooks: {
      afterAllFileWrite: () => {  
         // Do what you want here, all file is writed  
      },
    },
}
```

[a comment by an automated account is not shown]

**Reply by dotansimha** [reply]
you can find it here: 
https://github.com/dotansimha/graphql-code-generator/blob/e3af1dbc80a998444a1d1b353f127f297c6a09d0/packages/utils/plugins-helpers/src/types.ts#L533-L538

[a comment by an automated account is not shown]
